gtsam/linear/HessianFactor.cpp

```diff
@@ -32,14 +32,6 @@
 #include <boost/format.hpp>
 #include <boost/make_shared.hpp>
 #include <boost/tuple/tuple.hpp>
-#ifdef __GNUC__
```

Comment:
Same as above

Reply:
This was removed as I believe it was first introduced to surpress warnings from the previous `<boost/bind.hpp>` header. As we are not using that header, we can remove this `ifdef`.

Reply:
Even if we do use the older include, I believe the later versions of boost have fixed the associated warnings so there is no need to suppress them.